### PR TITLE
refactor: Avoid param reassignment in group_snippet

### PR DESCRIPTION
**Pull request by going-confetti**
## How to Test
⚠️ Please check that Validator works with different test scripts

## Checklist

- [x] I have performed a self-review of my code.
- [ ] I have added tests for my changes.
- [ ] I have commented on my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: resources/group_snippet.js [hunk 1]
```diff
@@ -45,44 +39,23 @@ class Client {
...
+function instrumentArguments(args) {
+  const [body = null, params = {}] = args
+  const groupHeaders = {
+    'X-k6-group': trimAndRemovePrefix(execution.vu.tags.group),
   }
 
-  return args
+  const updatedParams = Object.assign({}, params, {
+    headers: Object.assign({}, params.headers || {}, groupHeaders),
+  })
```
Comment (going-confetti) [inline resources/group_snippet.js:53]: TIL that our version of k6 doesn't support the spread operator. It was added to a newer version (https://github.com/grafana/k6/pull/3456)

File: resources/group_snippet.js
Comment (going-confetti) [inline resources/group_snippet.js:1]: I'm not quite happy with this snippet "polluting" the global namespace, essentially making it impossible to create user-defined variables with names like `Client`, `trimAndRemovePrefix`, etc - it's only possible in custom code snippets (and I'm still not sure whether those should have their own scope - let's have a separate discussion on that)

**Review by e-fisher — APPROVED** [review]
LGTM 👌

File: resources/group_snippet.js
Comment (Llandy3d) [reply resources/group_snippet.js:1]: good point!
We should definitely not reserve those names 🤔

**Pull request merged**